File: 189-rotate-array/189-rotate-array.java
class Solution {
public void rotate(int[] nums, int k) {
int n = nums.length;
k = k % n;
reverse(nums, 0, n - k - 1);
reverse(nums, n - k, n - 1);
reverse(nums, 0, n - 1);
}
public static void reverse(int[] nums, int i, int j){
int li = i;
int ri = j;
while(li < ri){
int temp = nums[li];
nums[li] = nums[ri];
nums[ri] = temp;
li++;
ri--;
}
}
}

File: 189-rotate-array/189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) :
        l = len(nums)
        k = k % l
        nums[:] = nums[(l-k):] + nums[:(l-k)]


#         for i in range(k):
#             for j in range(len(nums)-1,0,-1):
#                 nums[j],nums[j-1]  = nums[j-1],nums[j]
            
    
        
        
       